typings/atom-ide/index.d.ts

```diff
@@ -0,0 +1,267 @@
+import {Disposable, Grammar, Point, Range, TextEditor} from 'atom';
+
+/* declare module 'atom-ide' {*/
+
+  export type OutlineProvider = {
```

Comment:
This file is a conversion of the flowtype definitions for Atom-IDE and isn't finished yet.  There are some type system differences that I still need to account for.